File: leetcode/day8/python/so.py
from typing import List


class Solution:
    def findRepeatNumber(self, nums: List[int]) -> int:
        r = -1
        i = 0
        while i < len(nums):
            v = nums[i]
            if (v == i):
                i = i+1
                continue
            if (nums[v] == v):
                r = v
                break
            nums[v], nums[i] = nums[i], nums[v]
        return r

File: leetcode/day8/cpp/so.cpp
#include "so.h"
#include "iostream"

int Solution::findRepeatNumber(std::vector<int> &nums)
{

    for (size_t i = 0; i < nums.size();)
    {
        /* code */
        int val = nums[i];
        if (val == i)
        {
            i++;
            continue;
        }

        if (nums[val] == val)
        {
            return val;
        }

        nums[i] = nums[val];
        nums[val] = val;
    }

    return -1;
}